### PR TITLE
[BEAM-8648] Deprecate OutputHints from Euphoria API.

### PR DESCRIPTION
**Pull request by dmvk**
Deprecate OutputHints as they are no longer in use.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java [hunk 1]
```diff
@@ -186,10 +186,22 @@ default WindowedOutputBuilderT accumulatingFiredPanes() {
...
      *
-     * @param outputHints output dataset description
+     * @param outputHint output dataset description
+     * @param outputHints other output dataset descriptions
      * @return the dataset representing the new operator's output
+     * @deprecated Use {@link #output()} instead.
      */
-    PCollection<T> output(OutputHint... outputHints);
+    @Deprecated
+    default PCollection<T> output(OutputHint outputHint, OutputHint... outputHints) {
```
Comment (je-ik) [inline sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java]: Can we add a log here giving warnings that this call does effectively nothing now and will be removed?

File: sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java [hunk 2]
```diff
@@ -199,9 +211,22 @@ default WindowedOutputBuilderT accumulatingFiredPanes() {
      * Finalizes the operator and retrieves its output dataset. Using this output new operator
      * {@link MapElements} is added to the flow to extract values from pairs.
      *
-     * @param outputHints output dataset description
+     * @param outputHint output dataset description
+     * @param outputHints other output dataset descriptions
+     * @return the dataset representing the new operator's output
+     * @deprecated Use {@link #output()} instead.
+     */
+    default PCollection<V> outputValues(OutputHint outputHint, OutputHint... outputHints) {
```
Comment (je-ik) [inline sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java]: The same here.

File: sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java [hunk 3]
```diff
@@ -199,9 +215,25 @@ default WindowedOutputBuilderT accumulatingFiredPanes() {
...
+      LoggerFactory.getLogger(OutputValues.class)
+          .warn(
+              "OutputHints are deprecated and will be removed in next release. Use OutputValues#output() instead.");
```
Comment (je-ik) [inline sdks/java/extensions/euphoria/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java]:
```suggestion
              "OutputHints are deprecated and will be removed in next release. Use OutputValues#outputValues() instead.");
```

**Comment by je-ik** [comment]
Run Java PreCommit

**Comment by je-ik** [comment]
@dmvk what's missing here?

**Comment by je-ik** [comment]
@dmvk ping

**Comment by aaltay** [comment]
What should be the next action here? Should we close this PR? Is it ready to be merged?

**Comment by je-ik** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
Run Java PreCommit

**Comment by je-ik** [comment]
Closing in favor of #12388

**Pull request closed without merging**